apps/browser/src/autofill/services/autofill.service.ts

```diff
@@ -241,14 +241,18 @@ export default class AutofillService implements AutofillServiceInterface {
     }
 
     if (
-      cipher.reprompt !== CipherRepromptType.None &&
-      (await this.userVerificationService.hasMasterPasswordAndMasterKeyHash())
+      cipher.reprompt === CipherRepromptType.Password &&
+      (await this.userVerificationService.hasMasterPassword())
     ) {
       await BrowserApi.tabSendMessageData(tab, "passwordReprompt", {
         cipherId: cipher.id,
         action: "autofill",
       });
 
+      if (fromCommand) {
+        this.cipherService.updateLastUsedIndexForUrl(tab.url);
+      }
```

Comment:
```suggestion
      if (fromCommand) {
        this.cipherService.updateLastUsedIndexForUrl(tab.url);
      }
      
      await BrowserApi.tabSendMessageData(tab, "passwordReprompt", {
        cipherId: cipher.id,
        action: "autofill",
      });
```

As per our conversation, if we handle updating the cipher before the reprompt, we allow users the ability to close the reprompt window and cycle the next cipher. If the reprompt successfully goes through, then we treat that as the most recently filled element and remain at a status quo.